aws/solutions/TaggingRootVolumesInEC2/Tagging_Root_volume.yaml

```diff
@@ -0,0 +1,149 @@
+# * without restriction, including without limitation the rights to use, copy, modify,
+# * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+# * permit persons to whom the Software is furnished to do so.
+# *
+# * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+# * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+# * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+# */
+
+AWSTemplateFormatVersion: 2010-09-09
+Description: 'AWS CloudFormation Sample Template Tagging Root Volumes of EC2 Instances:
+  This template shows how to automatically tag the root volume of the EC2 instances which are
+  created through the CloudFormation template. This is done through the UserData property of 
+  the AWS::EC2::Instance resource.
+  **WARNING** This template creates two Amazon EC2 instances and an IAM Role.
+  You will be billed for the AWS resources used if you create a stack from this template.'
+Parameters:
+  KeyName:
+    Type: AWS::EC2::KeyPair::KeyName
+    Description: Name of an existing EC2 KeyPair to enable SSH access to the ECS instances.
+  InstanceType:
+    Description: EC2 instance type
+    Type: String
+    Default: t2.micro
+    AllowedValues: [t2.micro, t2.small, t2.medium, t2.large, m3.medium, m3.large,
+      m3.xlarge, m3.2xlarge, m4.large, m4.xlarge, m4.2xlarge, m4.4xlarge, m4.10xlarge,
+      c4.large, c4.xlarge, c4.2xlarge, c4.4xlarge, c4.8xlarge, c3.large, c3.xlarge,
+      c3.2xlarge, c3.4xlarge, c3.8xlarge, r3.large, r3.xlarge, r3.2xlarge, r3.4xlarge,
+      r3.8xlarge, i2.xlarge, i2.2xlarge, i2.4xlarge, i2.8xlarge]
+    ConstraintDescription: Please choose a valid instance type.
+  InstanceAZ:
+   Description: EC2 AZ.
+   Type: AWS::EC2::AvailabilityZone::Name
+   ConstraintDescription: "Must be the name of an availabity zone."
+  WindowsAMIID:
+    Description: The Windows 2016 AMIID, the default value is for the eu-west-1 region
+    Type: String
+    Default: ami-0d138b26f46625e2f
+  LinuxAMIID:
+    Description: The Amazon Linux 2 AMIID, the default value is for the eu-west-1 region
+    Type: String
+    Default: ami-0fad7378adf284ce0
+
+Resources:
+  #Tagging the root volume for the EC2 instances with the Windows OS
+  #The commented out powershell commands are to utilize if the AMI has the AWS CLI installed and you prefer that approach.
+  WindowsInstance:
+    Type: 'AWS::EC2::Instance'
+    Properties:
+      ImageId: !Ref WindowsAMIID
+      InstanceType: !Ref InstanceType
+      AvailabilityZone: !Ref InstanceAZ
+      IamInstanceProfile: !Ref InstanceProfile
+      KeyName: !Ref KeyName
+      UserData: 
+        Fn::Base64: !Sub |

```

Comment:
You have a !Sub for your user-data, but you're not doing parameter injection at the CloudFormation level -- is this by design?